failed to pull-based replication from remote harbor

2020-12-13T10:59:36Z [ERROR] [/replication/transfer/image/transfer.go:163]: http status code: 401, body: {"errors":[{"code":"UNAUTHORIZED","message":"unauthorized to access repository: goharbor/harbor-operator, action: push: unauthorized to access repository: goharbor/harbor-operator, action: push"}]}
![image](https://user-images.githubusercontent.com/188115/102009911-97ab8380-3d75-11eb-9b01-3d1cbb75d312.png)
![image](https://user-images.githubusercontent.com/188115/102009933-b4e05200-3d75-11eb-8274-ab4ee4e9fc6b.png)
![image](https://user-images.githubusercontent.com/188115/102009985-08eb3680-3d76-11eb-8acc-2101ba6f457f.png)

--- a/controllers/goharbor/harbor/jobservice.go
+++ b/controllers/goharbor/harbor/jobservice.go
@@ -106,9 +106,16 @@ func (r *Reconciler) GetJobService(ctx context.Context, harbor *goharborv1alpha2
 	name := r.NormalizeName(ctx, harbor.GetName())
 	namespace := harbor.GetNamespace()
 
+	host := r.NormalizeName(ctx, harbor.GetName(), controllers.Core.String())
+	if harbor.Spec.InternalTLS.IsEnabled() {
+		host += ":443"
+	} else {
+		host += ":80"
+	}
+
 	coreURL := (&url.URL{
 		Scheme: harbor.Spec.InternalTLS.GetScheme(),
-		Host:   r.NormalizeName(ctx, harbor.GetName(), controllers.Core.String()),
+		Host:   host,
 	}).String()
 	coreSecretRef := r.NormalizeName(ctx, harbor.GetName(), controllers.Core.String(), "secret")
 	registryAuthRef := r.NormalizeName(ctx, harbor.GetName(), controllers.Registry.String(), "basicauth")

